mava/networks/torsos.py

```diff
@@ -76,6 +76,19 @@ def __call__(self, observation: chex.Array) -> chex.Array:
         return jax.lax.collapse(x, -3)
 
 
+class SwiGLU(nn.Module):
+    ffn_dim: int
+    embed_dim: int
+
+    def setup(self) -> None:
+        self.W_1 = self.param("W_1", nn.initializers.zeros, (self.embed_dim, self.ffn_dim))
+        self.W_G = self.param("W_G", nn.initializers.zeros, (self.embed_dim, self.ffn_dim))
+        self.W_2 = self.param("W_2", nn.initializers.zeros, (self.ffn_dim, self.embed_dim))
+
+    def __call__(self, x: chex.Array) -> chex.Array:
+        return (jax.nn.swish(x @ self.W_G) * (x @ self.W_1)) @ self.W_2
+
```

Comment:
```suggestion
class SwiGLU(nn.Module):
    """SiwGLU module for Sable's Network.

    Implements the SwiGLU feedforward neural network module, which is a variation
    of the standard feedforward layer using the Swish activation function combined
    with a Gated Linear Unit (GLU).
    """

    hidden_dim: int
    input_dim: int

    def setup(self) -> None:
        # Initialize the weights for the SwiGLU layer
        self.W_linear = self.param(
            "W_linear", nn.initializers.zeros, (self.input_dim, self.hidden_dim)
        )
        self.W_gate = self.param("W_gate", nn.initializers.zeros, (self.input_dim, self.hidden_dim))
        self.W_output = self.param(
            "W_output", nn.initializers.zeros, (self.hidden_dim, self.input_dim)
        )

    def __call__(self, x: chex.Array) -> chex.Array:
        """Applies the SwiGLU mechanism to the input tensor `x`."""
        # Apply Swish activation to the gated branch and multiply with the linear branch
        gated_output = jax.nn.swish(x @ self.W_gate) * (x @ self.W_linear)
        # Transform the result back to the input dimension
        return gated_output @ self.W_output

```

Reply:
I added documentation and updated vars naming in swiglu, i added this suggestion here since MAT PR will be first to be merged, also adding it to torsos.py is better than in the outside utils folder (i will update sable PR based on that)

Reply:
Thank you for the suggestions! I implemented some of them and then didn't add some because I feel the code is self-explanatory enough. 
I also think naming the one variable `embed_dim` instead of `input_dim` is more consistent with the rest of our code.